cmake/compiler/gcc/target.cmake

```diff
@@ -54,6 +54,10 @@ elseif("${ARCH}" STREQUAL "riscv")
   else()
     list(APPEND TOOLCHAIN_C_FLAGS -mabi=ilp32 -march=rv32ima)
   endif()
+elseif("${ARCH}" STREQUAL "x86")
+  include(${ZEPHYR_BASE}/cmake/compiler/gcc/target_x86.cmake)
```

Comment:
Could you try using CMAKE_CURRENT_SOURCE_DIR or CMAKE_CURRENT_LIST_DIR  ?

Reply:
`CMAKE_CURRENT_LIST_DIR` works.